src/window.c

```diff
@@ -6436,7 +6435,11 @@ win_fix_cursor(int normal)
 	}
 	else
 	{   // Ensure cursor stays visible if we are not in normal mode.
-	    wp->w_fraction = top ? 0 : FRACTION_MULT;
+	    wp->w_fraction = 0.5 * FRACTION_MULT;
+	    // Make sure cursor is closer to topline than botline.
+	    if (so == wp->w_height / 2
+			  && nlnum - wp->w_topline > wp->w_botline - 1 - nlnum)
+		wp->w_fraction += 1L;
```

Comment:
I think this will be covered by the test if we increase the loop range. Should we? Not sure how to determine what range will cover it. I tested it on my local system and confirmed this fixes a problem but when this happens depends on the window height the test runs in.